src/_path.h

```diff
@@ -821,8 +822,22 @@ inline bool segments_intersect(const double &x1,
                                const double &x4,
                                const double &y4)
 {
+    // determinant
     double den = ((y4 - y3) * (x2 - x1)) - ((x4 - x3) * (y2 - y1));
-    if (den == 0.0) {
+    if (den == 0.0) {  // collinear segments
+        if (x1 == x2 && x2 == x3) { // segments have infinite slope (vertical lines)
+                                    // and lie on the same line
+            return (fmin(y1, y2) <= fmin(y3, y4) && fmin(y3, y4) <= fmax(y1, y2)) ||
+                   (fmin(y3, y4) <= fmin(y1, y2) && fmin(y1, y2) <= fmax(y3, y4));
+        }
+        else {
+            double intercept = (y1*x2 - y2*x1)*(x4 - x3) - (y3*x4 - y4*x3)*(x1 - x2);
+            if (intercept == 0.0) { // segments lie on the same line
+                return (fmin(x1, x2) <= fmin(x3, x4) && fmin(x3, x4) <= fmax(x1, x2)) ||
+                       (fmin(x3, x4) <= fmin(x1, x2) && fmin(x1, x2) <= fmax(x3, x4));
+            }
```

Comment:
This looks better to me, though I didn't check your math.  Note that you could just have the one return statement using y only if you wanted it all to be more elegant.    Thanks for sticking with this!

Reply:
I don't think we can use `y`'s only because of the same reasoning why we cannot use only `x`'s.

Reply:
If `intercept == 0`, it doesn't matter if you check y or x, everything is on the same line...

Reply:
In the `intercept == 0`, we excluded `x = const` lines, but if we replace `x`'s with `y`'s in the return statement the code will fail for `y = const` lines.

Reply:
Ah, OK, fair enough...  Thanks!